packages/ckeditor5-link/src/linkimageui.js

```diff
@@ -34,6 +40,91 @@ export default class LinkImageUI extends Plugin {
 		return 'LinkImageUI';
 	}
 
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * The plugin button view.
+		 *
+		 * @member {module:ui/button/buttonview~Button}
+		 */
+		this.linkButtonView = null;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
 	init() {
+		const editor = this.editor;
+		const viewDocument = editor.editing.view.document;
+
+		this.listenTo( viewDocument, 'click', ( evt, data ) => {
+			const hasLink = isImageLinked( editor.model.document.selection.getSelectedElement() );
+
+			if ( hasLink ) {
+				data.preventDefault();
+			}
+		} );
+
+		this._createToolbarLinkImageButton();
 	}
+
+	/**
+	 * Creates a LinkImageUI button view.
+	 * Clicking on the button shows a {@link module:link/linkui~LinkUI#_balloon} attached to the selection.
+	 * When image is already linked, the view shows {@link module:link/linkui~LinkUI#actionsView} or
+	 * {@link module:link/linkui~LinkUI#formView} if it's not.
+	 *
+	 * @private
+	 */
+	_createToolbarLinkImageButton() {
+		const editor = this.editor;
+		const t = editor.t;
+
+		editor.ui.componentFactory.add( 'linkImage', locale => {
+			const button = new ButtonView( locale );
+			const plugin = editor.plugins.get( 'LinkUI' );
+			const linkCommand = editor.commands.get( 'link' );
+
+			button.set( {
+				isEnabled: true,
+				label: t( 'Link image' ),
+				icon: linkIcon,
+				keystroke: linkKeystroke,
+				tooltip: true,
+				isToggleable: true
+			} );
+
+			// Bind button to the command.
+			button.bind( 'isEnabled' ).to( linkCommand, 'isEnabled' );
+			button.bind( 'isOn' ).to( linkCommand, 'value', value => !!value );
+			button.bind( 'hasLink' ).to( this, 'hasLink' );
+
+			// Show the actionsView or formView (both from LinkUI) on button click depending on whether the image is linked already.
+			this.listenTo( button, 'execute', () => {
+				const hasLink = isImageLinked( editor.model.document.selection.getSelectedElement() );
+
+				hasLink ? plugin._addActionsView() : plugin._showUI( true );
```

Comment:
`if {} else {}` please unless the result is an argument of a method e.g. `foo( isForward ? 'forward' : 'backward' )`